packages/codegen-ui-react/lib/forms/form-state.ts

```diff
@@ -120,7 +120,12 @@ export const getDefaultValueExpression = (
   if (componentType in componentTypeToDefaultValueMap) {
     return componentTypeToDefaultValueMap[componentType];
   }
-  return factory.createIdentifier('undefined');
+  /**
+   * All other types are strings.
+   * Use an empty string here instead of undefined because undefined will cause the component to go from uncontrolled
+   * to controlled. A component should be controlled or uncontrolled for the lifetime of the component.
+   */
+  return factory.createStringLiteral('');
```

Comment:
Previous update called this function to get defaultValues. But I had to update this to return an empty string for all other types.